src/cast_core/src/CollectionScanner.cpp

```diff
@@ -219,15 +234,22 @@ void collectionScan(CollectionScanner::PhaseConfig* config,
     bool scanFinished = false;
     auto statTracker = config->scanOperation.start();
     for (auto& collection : collections) {
-        auto filter = config->filterExpr ? config->filterExpr->evaluate()
-                                         : bsoncxx::document::view_or_value{};
-        auto docs = collection.find(session, filter, config->findOptions);
+        auto docs = [&]() {
+          if (config->aggregate) {
+              return collection.aggregate(session, evaluatePipeline(config), config->aggregateOptions);
+          } else {
```

Comment:
You do one or the other here. I think we should either support running both, or the constructor should throw an error if you set both.

Reply:
It should throw an exception. I'll fix that.